packages/shared_preferences/shared_preferences_platform_interface/lib/method_channel_shared_preferences.dart

```diff
@@ -0,0 +1,225 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+import 'package:meta/meta.dart';
+
+import 'shared_preferences_platform_interface.dart';
+
+const MethodChannel _kChannel =
+    MethodChannel('plugins.flutter.io/shared_preferences');
+
+const String _prefix = 'flutter.';
+
+class MethodChannelSharedPreferencesBinding
+    implements SharedPreferencesBinding {
```

Comment:
So the way that others are going to use this is by doing:

```dart
class WebSharedPreferencesBinding implements SharedPreferencesBinding {

@override
static SharedPreferencesBinding instance =
      WebSharedPreferencesBinding();

...
```

Unless the interface of the `SharedPreferencesBinding` never changes, doesn't this need to be `extends` so changes in the interface don't cause compile-time errors in other packages?

IIRC extends is used in the PlatformInterface packages so they act similarly to Java "interfaces with default implementations".